### DISCUSSION TITLE
How to create requirement.txt?

### DISCUSSION THREAD

**Discussion by yalunren** [astral-sh/rye#499] [General]
Since there is no pip in rye, how to create a requirement.txt? Currently I manually copy and change requirement.lock file, and remove headings, which isn't very convienent and can cause error sometimes. Isn't there a command that can do it?

Thanks!

**Comment by smokeythemonkey** [comment]
See: https://github.com/mitsuhiko/rye/discussions/239#discussioncomment-6032595

you can define `sed '/-e/d' requirements.lock > requirements.txt` as a script in your pyproject.toml

**Comment by impredicative** [comment]
By `requirements.txt` do you mean the top-level requirements or all (including transitive) requirements? The top-level ones can be obtained from `pyproject.toml`. All requirements, including transitive ones, can be obtained from `requirements.lock` and/or `requirements-dev.lock` depending on specifically what you need. 

Lastly, note that I see the non-dev ones in `requirements-dev.lock` too, which I consider to be an unusual and unnecessary duplication.
